### PR TITLE
fix(argo-workflows): Add unhealthyPodEvictionPolicy to workflow controller PDB

### PR DESCRIPTION
**Pull request by fleuryc-owkin**
Resolves https://github.com/argoproj/argo-helm/issues/3011

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-workflows/templates/controller/workflow-controller-deployment-pdb.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ spec:
   {{- else }}
   minAvailable: 0
   {{- end }}
+  unhealthyPodEvictionPolicy: IfHealthyBudget
```
Comment (tico24) [inline charts/argo-workflows/templates/controller/workflow-controller-deployment-pdb.yaml:17]: Feels like this should be user-configurable as some users may well prefer `AlwaysAllow`

**Review by tico24 — CHANGES_REQUESTED** [review]

**Comment by tico24** [comment]
Also this only went stable fairly recently didn't it (maybe 1.31 from memory)? How will this work for older k8s clusters?

File: charts/argo-workflows/templates/controller/workflow-controller-deployment-pdb.yaml (hunk 1 above)
Comment (fleuryc-owkin) [reply charts/argo-workflows/templates/controller/workflow-controller-deployment-pdb.yaml]: Done ✅ 

I ran `./scripts/helm-docs.sh` but the attribute doesn't show up in the doc. What did I do wrong ?

File: charts/argo-workflows/values.yaml [hunk 2]
```diff
@@ -344,6 +344,9 @@ controller:
     enabled: false
     # minAvailable: 1
     # maxUnavailable: 1
+    
+    # -- [unhealthyPodEvictionPolicy] is only valid for v1.31+
```
Comment (tico24) [inline charts/argo-workflows/values.yaml:348]: v1.31 of what?
It's valid for earlier versions but it's behind a gate.

Comment (fleuryc-owkin) [reply charts/argo-workflows/values.yaml]: Fixed the comment ✅

File: charts/argo-workflows/Chart.yaml [hunk 3]
```diff
@@ -18,3 +18,5 @@ annotations:
   artifacthub.io/changes: |
     - kind: added
       description: Support lifecycle and terminationGracePeriodSeconds for server
```
Comment (yu-croco) [inline charts/argo-workflows/Chart.yaml:20]: Please remove old one.

**Pull request closed without merging**